### ISSUE TITLE
About balance loss

### ISSUE THREAD

**Issue by JiawangLiu** [laekov/fastmoe#128]
hi, if I want to use balance loss in my transformer model, what's the mean of `clear` in the `get_loss` method, and when should I set it to `True` or `False`. Besides, `gate hook` should be used in my code? Thank you!

**Comment by laekov** [comment]
By default, we assume that `get_loss` is only called once in each itreation. Thus, `get_loss` by default transfers ownership of the loss tensor to the user, and the whole backward graph with intermediate tensors can be released as soon as possible.

`clear` can be set to `False` if you want to have the loss for debug or other purpose that does not want to drop the loss in the gate instance. `gate_hook` is for similar debugging purpose that you can have the gate's output in each iteration in your customized gate hook function.

**Comment by JiawangLiu** [comment]
In my code, I get the balance_loss by the follows and no `gate_hook` used, is this the correct usage?
```python
balance_loss=([layer.moe_ff.gate.get_loss(clear=True) for layer in self.encoder.layers if layer.moe_ff.gate.has_loss])
balance_loss = torch.stack(balance_loss).mean() * 12
loss += balance_loss
```

**Comment by laekov** [comment]
That looks good
